### PR TITLE
Add showCancelled to ListAllEvents and filter on backend

### PR DESCRIPTION
**Pull request by nabramow**
This adds `showCancelled` to the `ListAllEvents` function on the backend, to the protos and switches out the existing events frontend to filter on the backend instead of the frontend (same as `ListMyEvents` works).

**TO TEST:**

- Regenerate your protos locally and restart your docker containers on the backend
- Go to the main events page
- Make sure you have a canceled event in the events list. Create an event and cancel it if not.
- Toggle the "show cancelled". It should add and remove the cancelled events from the list.

**Backend checklist**
- [ x ] Formatted my code by running `ruff check --select I --fix . && ruff check . && ruff format .` in `app/backend`
- [ x ] Added tests for any new code or added a regression test if fixing a bug
- [ x ] All tests pass
- [ x ] Run the backend locally and it works
- [ N/A ] Added migrations if there are any database changes, rebased onto `develop` if necessary for linear migration history

**Web frontend checklist**
- [ x ] Formatted my code with `yarn format`
- [ x ] There are no warnings from `yarn lint --fix`
- [ x ] There are no console warnings when running the app
- [ N/A ] Added any new components to storybook
- [ x ] Added tests where relevant
- [ x ] All tests pass
- [ x ] Clicked around my changes running locally and it works
- [ x ] Checked Desktop, Mobile and Tablet screen sizes

### CODE REVIEW

**Comment by aapeliv** [comment]
Try this hash: `9e66f9be792b3f27b73c6d8f7b2150f84406948edca85252134fc212a3fd14e8`

File: app/proto/events.proto [hunk 1]
```diff
@@ -294,6 +294,8 @@ message ListAllEventsReq {
 
   uint32 page_size = 2;
   string page_token = 3;
+  
+  bool include_cancelled = 4;
```
Comment (aapeliv) [inline app/proto/events.proto]: Please move this above the `page_{size,token}`

**Review by aapeliv — COMMENTED** [review]
One minor comment.

File: app/proto/events.proto (hunk 1 above)
Comment (nabramow) [reply app/proto/events.proto]: Done! Just out of curiosity, what does moving the location up do?

Comment (aapeliv) [reply app/proto/events.proto]: Nothing, but it's just clearer since the pagination stuff should come after all the content :)

**Review by aapeliv — DISMISSED** [review]
LGTM

**Review by aapeliv — APPROVED** [review]

**Pull request merged**